packages/shared/src/components/cards/v1/ArticlePostCard.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { forwardRef, ReactElement, Ref } from 'react';
+import classNames from 'classnames';
+import Link from 'next/link';
+import { CardLink, CardTextContainer, CardTitle } from './Card';
+import ActionButtons from './ActionButtons';
+import { PostCardHeader } from './PostCardHeader';
+import { PostCardFooter } from './PostCardFooter';
+import { Container, PostCardProps } from '../common';
+import FeedItemContainer from './FeedItemContainer';
+import { useBlockPostPanel } from '../../../hooks/post/useBlockPostPanel';
+import { PostTagsPanel } from '../../post/block/PostTagsPanel';
+import { useFeedPreviewMode, usePostFeedback } from '../../../hooks';
+import { FeedbackCard } from './FeedbackCard';
+import { Origin } from '../../../lib/analytics';
+
+export const ArticlePostCard = forwardRef(function PostCard(
+  {
+    post,
+    onPostClick,
+    onUpvoteClick,
+    onDownvoteClick,
+    onCommentClick,
+    onMenuClick,
+    onBookmarkClick,
+    onShareClick,
+    openNewTab,
+    children,
+    showImage = true,
+    insaneMode,
+    onReadArticleClick,
+    domProps = {},
+  }: PostCardProps,
+  ref: Ref<HTMLElement>,
+): ReactElement {
+  const { className, style } = domProps;
+  const { data } = useBlockPostPanel(post);
+  const { type, pinnedAt, trending } = post;
+
+  const onPostCardClick = () => onPostClick?.(post);
+
+  const customStyle = !showImage ? { minHeight: '15.125rem' } : {};
+  const { showFeedback } = usePostFeedback({ post });
+  const isFeedPreview = useFeedPreviewMode();
+
+  if (data?.showTagsPanel && post.tags.length > 0) {
+    return (
+      <PostTagsPanel className="overflow-hidden" post={post} toastOnSuccess />
+    );
+  }
+
+  return (
+    <FeedItemContainer
+      domProps={{
+        ...domProps,
+        style: { ...style, ...customStyle },
+        className,
+      }}
+      ref={ref}
+      flagProps={{ pinnedAt, trending, type }}
+    >
+      {!isFeedPreview && (
+        <Link href={post.commentsPermalink}>
+          <CardLink
+            title={post.title}
+            onClick={onPostCardClick}
+            href={post.commentsPermalink}
+          />
+        </Link>
+      )}
+
+      {showFeedback ? (
+        <FeedbackCard
+          post={post}
+          onUpvoteClick={() => onUpvoteClick(post, Origin.FeedbackCard)}
+          onDownvoteClick={() => onDownvoteClick(post, Origin.FeedbackCard)}
+          showImage={showImage}
```

Comment:
I know this is not part of this task, but since the changes are here, trying to understand why the change of logic on how we show the feedback card?

Reply:
It's mostly the implementation is cleaner this way. Instead of hiding components with CSS we do not even render it, if it's not shown.